bigbluebutton-html5/imports/ui/components/video-provider/component.jsx

```diff
@@ -192,16 +94,18 @@ class VideoDock extends Component {
     }
 
     Object.keys(this.webRtcPeers).forEach((id) => {
-     this.destroyWebRTCPeer(id);
+      this.destroyWebRTCPeer(id);
     });
+
     // Close websocket connection to prevent multiple reconnects from happening
     this.ws.close();
   }
 
-  adjustVideos() {
-    setTimeout(() => {
-      window.adjustVideos('webcamArea', true, mediaStyles.moreThan4Videos, mediaStyles.container, mediaStyles.overlayWrapper, 'presentationAreaData', 'screenshareVideo');
-    }, 0);
+  componentWillUpdate(nextProps) {
+    const { isLocked } = nextProps;
+    if (isLocked && VideoService.isConnected()) {
+      this.unshareWebcam();
```

Comment:
I still dont understand why you have unshare here. You should control shared/unshared via mount/unmount methods on the VideoElement.